src/main/java/com/puppycrawl/tools/checkstyle/DefaultLogger.java

```diff
@@ -76,9 +77,22 @@ public class DefaultLogger extends AutomaticBean implements AuditListener {
      * @param outputStream where to log audit events
      * @param outputStreamOptions if {@code CLOSE} that should be closed in auditFinished()
      */
-    public DefaultLogger(OutputStream outputStream, OutputStreamOptions outputStreamOptions) {
+    public DefaultLogger(OutputStream outputStream,
+                         AutomaticBean.OutputStreamOptions outputStreamOptions) {
+        this(outputStream, OutputStreamOptions.values()[outputStreamOptions.ordinal()]);
+    }
```

Comment:
@rnveach This is the enum conversion in action. We only need to do it here and in `XMLLogger`, so I think it is okay like that. Or maybe make some util function though it would be temporary until `AutomaticBean` gets removed. What do you think?

Taken from [this StackOverflow answer](https://stackoverflow.com/a/7285219/9553927).

---
Do you have any ideas on how to tackle [the new surviving mutation](https://github.com/checkstyle/checkstyle/actions/runs/3453042315/jobs/5763319036#step:7:16):
```
 Source File: "DefaultLogger.java"
Class: "com.puppycrawl.tools.checkstyle.DefaultLogger"
Method: "&lt;init&gt;"
Line Contents: "this(outputStream, OutputStreamOptions.values()[outputStreamOptions.ordinal()]);"
Mutator: "org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator"
Description: "removed call to com/puppycrawl/tools/checkstyle/api/AutomaticBean$OutputStreamOptions::ordinal"
Line Number: 82
``` 
`ordinal()` is essentially swapped with `0` so we get `this(outputStream, OutputStreamOptions.values()[0]);`

Reply:
Fields that get set depending on value of `outputStreamOptions` aren't public so we cannot just check them and fail a test.

Reply:
> Do you have any ideas on how to tackle [the new surviving mutation](https://github.com/checkstyle/checkstyle/actions/runs/3453042315/jobs/5763319036#step:7:16):

I would check out https://github.com/checkstyle/checkstyle/issues/12341 , but I don't seeing regression being helpful for this case where the target of the regression is basically infrastructure related.

You will need to examine the code and the mutation being applied to understand how to kill it.

> ordinal

This is such odd code.

Reply:
Made a temporary public method `isCloseStream()` to kill the mutation.

> This is such odd code.

Okay, casting by shared enum value:
```
this(outputStream, OutputStreamOptions.valueOf(outputStreamOptions.name()));
```

Done.